### ISSUE TITLE
mat-select not showing mat-error on submit. 

### ISSUE THREAD

**Issue by RenaldoV** [angular/material2#8271]
#### Bug, feature request, or proposal:
mat-error not displayed when form is submitted. I see there was an issue on this which has been closed, but I'm still experiencing this problem after my update to 2.00-beta-12. 

```
<mat-form-field>
      <mat-select placeholder="Choose Job Role*" formControlName="role" ng-required="true">
        <mat-option *ngFor="let rl of roles" [value]="rl">{{ rl }}</mat-option>
      </mat-select>
      <mat-error *ngIf="role.hasError('required')">
        Job Role is required.
      </mat-error>
</mat-form-field>
```

on touch and leave the error shows, but not on submit.  

#### What is the expected behavior?
After form submission the relevant error message should show if the select field is invalid.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

* Angular ^4.4.6
    * was on ^5.00 but mat-select threw error when changing the chosen option so on https://github.com/angular/material2/issues/8213 #issuecomment-341735433 they said to downgrade core to 4.x and so I did to avoid  getting the error.
* Typescript ^2.6.1
* Material ^2.00-beta.12
* OS Windows 10
* Browsers (Chrome, Firefox)

**Comment by willshowell** [comment]
Fixed in rc0 via https://github.com/angular/material2/pull/7640.

https://stackblitz.com/edit/material2-rc0-tcjgdf?file=app%2Fapp.component.html

**Comment by josephperrott** [comment]
Closing as closed by #7640
